### DISCUSSION TITLE
use module $params in data.php (dynamicGet)

### DISCUSSION THREAD

**Discussion by saman222** [joomengine/Joomla-Component-Builder#656] [Ideas]
hi.
how to access $params variable in data.php which is accessible in mod_modulename.php of module?
this capability need for use module params (which set in admin page of module in joomla) as input argument for daynamicGet function witch is set in the Component menu>component builder>joomla module>code tab> Custom Gets

this idea come from:
https://github.com/vdm-io/Joomla-Component-Builder/discussions/654#discussioncomment-295756

**Comment by Llewellynvdm** [comment]
Okay so I can add an area that will allow you to pass values to the class during initializing like so:
```php
Mod[[[Module]]]Data = new Mod[[[Module]]]Data($params);
$values = Mod[[[Module]]]Data->getValues();
```
So you can via the constructor populate the data class. The data class is buildup of all the custom gets you add, so passing it as a param to the constructor will make it available to any of the methods.

**Reply by Llewellynvdm** [reply]
The new **area** will allow you to add any other custom method including the constructor method the the data class. Also note that the data class is extending the JObject class at this time, so it has the setter and getter methods in-place and ready for use.

**Reply by saman222** [reply]
excuse me @Llewellynvdm 
maybe my question is silly but where i can use constructor? where must write constructor function? can you give me a simple example or any advise? are you implement new **area** which you say in JCB?

**Reply by ArMaNKashef** [reply]
Hi
@Llewellynvdm , @saman222 
you can pass any variable from mod_yourmodule.php to data.php like this:
In  mod_yourmodule.php :
`
/* for example your variable is $x */
$app = JFactory::getApplication();
$app->input->set('key_to_x', $x);
`
In data.php:
`
$this->app = JFactory::getApplication();
$this->input = $this->app->input;
$x = $this->input->getArray()["key_to_x"];
`
This worked for me.

**Reply by Llewellynvdm** [reply]
The new area does not yet exist in JCB (module) view. But I am planning to add it. Which will in fact give us more control over the data class, since now we can only add custom get methods to this class. Yet with my planed change we will be able to add our own code methods, like the constructor ;)

@ArMaNKashef the module $params are not part of the input values, as they are set in the module admin area.
![image](https://user-images.githubusercontent.com/5607939/105322783-ffa18500-5bd1-11eb-8bb1-624f60b95951.png)
